pkg/dotc1z/c1file.go

```diff
@@ -412,3 +455,70 @@ func (c *C1File) GrantStats(ctx context.Context, syncType connectorstore.SyncTyp
 
 	return stats, nil
 }
+
+// isWALMode checks if the database is using WAL mode.
+func (c *C1File) isWALMode(ctx context.Context) bool {
+	for _, pragma := range c.pragmas {
+		if pragma.name == "journal_mode" && strings.EqualFold(pragma.value, "wal") {
+			return true
+		}
+	}
+
+	var mode string
+	if err := c.rawDb.QueryRowContext(ctx, "PRAGMA journal_mode").Scan(&mode); err == nil {
+		return strings.EqualFold(mode, "wal")
+	}
+
+	return false
+}
+
+// startWALCheckpointing starts a background goroutine to perform WAL checkpoints every 5 minutes.
+func (c *C1File) startWALCheckpointing() {
+	c.checkpointTicker = time.NewTicker(5 * time.Minute)
```

Comment:
Should `WithWALCheckpointDuration(time.Duration)` be the option instead of (or maybe in addition to) `WithC1FWALCheckpoint(bool)`? (If it's instead of, a duration of 0 would disable.) I'm guessing we'll want to tweak this duration based on what we learn from running it in prod.